helix-term/src/handlers/completion.rs

```diff
@@ -291,6 +302,200 @@ fn request_completion(
     });
 }
 
+fn path_completion(
+    cursor: usize,
+    text: helix_core::Rope,
+    doc: &helix_view::Document,
+) -> Option<BoxFuture<'static, anyhow::Result<Vec<CompletionItem>>>> {
+    if !doc.supports_path_completion() {
+        return None;
+    }
+
+    // TODO find a good regex for most use cases (especially Windows, which is not yet covered...)
+    // currently only one path match per line is possible in unix
+    static PATH_REGEX: Lazy<rope::Regex> =
+        Lazy::new(|| rope::Regex::new(r"((?:~|\$HOME|\$\{HOME\})?(?:\.{0,2}/)+.*)$").unwrap());
+
+    let cur_line = text.char_to_line(cursor);
+    let start = text.line_to_char(cur_line).max(cursor.saturating_sub(1000));
+    let line_until_cursor = text.slice(start..cursor).regex_input_at(..);
+
+    let (dir_path, typed_file_name) = PATH_REGEX.search(line_until_cursor).and_then(|m| {
+        let start_byte = text.char_to_byte(start);
+        let matched_path = &text
+            .byte_slice((start_byte + m.start())..(start_byte + m.end()))
+            .to_string();
+
+        // resolve home dir (~/, $HOME/, ${HOME}/) on unix
+        #[cfg(unix)]
+        let mut path = {
+            static HOME_DIR: Lazy<Option<OsString>> = Lazy::new(|| std::env::var_os("HOME"));
+
+            let home_resolved_path = if let Some(home) = &*HOME_DIR {
+                let first_separator_after_home = if matched_path.starts_with("~/") {
```

Comment:
I think for this feature in particular (but also gf) proper path expansion would be nice instead of this special case.